core/src/scss/utilities/mixins/link/_link-icon.scss

```diff
@@ -9,37 +9,83 @@
 /// @param {string} $width  - Width of icon including unit, e.g. 0.75em, 24px.
 /// @param {string} $animate  - Type of animation for the icon on hover. Available options: right, down, up, topright, none. (deprecated true. Please use right)
 /// @param {string} $vertical  - Adjustment of vertical position of icon, e.g., '2px' moves icon up 2 pixels, '-0.3em' moves it down 0.3 em.
+/// @param {string} $location - Location of icon, e.g., before or after.
 ///
 /// @group mixin
-@mixin link-icon($icon, $width: 0.65em, $animate: right, $vertical: null) {
+@mixin link-icon(
+  $icon,
+  $width: 0.65em,
+  $animate: right,
+  $vertical: null,
+  $location: after
+) {
   text-decoration: none;
   @supports (mask-repeat: no-repeat) {
-    &::after {
-      @include margin(null 0.3em $vertical 0.4em);
+    // Common properties for both before and after locations
+    &::#{$location} {
       @include size($width);
       display: inline-block;
       content: '';
-      mask: url("#{$su-image-path}/#{$icon}.svg") no-repeat 0 0;
+      mask: url('#{$su-image-path}/#{$icon}.svg') no-repeat 0 0;
       mask-size: contain;
-      background-color: color(link);
-      transition: transform 0.2s ease-in-out, mask-image 0.2s ease-in-out;
     }
 
     &:hover,
     &:focus {
-      &::after {
-        background-color: color(link--hover);
-        @if $animate == 'right' or $animate == 'true' {
-          transform: translateX(0.2em);
-        }
-        @else if $animate == 'down' {
+      &::#{$location} {
+        @if $animate == 'down' {
           transform: translateY(0.2em);
         }
         @else if $animate == 'up' {
           transform: translateY(-0.2em);
         }
-        @else if $animate == 'topright' {
-          transform: translate3d(0.15em, -0.15em, 0);
+      }
+    }
+
+    // Icon $location after link text
+    @if $location == 'after' {
+      &::after {
+        @include margin(null 0.1em $vertical 0.1em);
+        background-color: color(link);
+
+        @if $animate == 'right' or $animate == 'true' or $animate == 'topright' {
+          @include margin(null 0.3em $vertical 0.4em);
```

Comment:
@cjwest it looks like here you modified the plugin to add smaller horizontal margin to the icon if there's no animation (0.1em on both left and right), but add more margin if there is animation. I'm a little worried that this change will cascade, since for example UComm might be using the non animated version of this mixin 🤔